tests/test_file_save_gui.py

```diff
@@ -0,0 +1,364 @@
+    app_dialog.root.outlineitems["Model"].waitExist(timeout=30)
+
+    # Validate content dialog
+    assert app_dialog.root.cells[
+        "AssetAutomation"
+    ].exists(), "AssetAutomation is missing in content dialog"
+    assert app_dialog.root.cells[
+        "Model - Model"
+    ].exists(), "Model task is missing in content dialog"
+    assert app_dialog.root.cells[
+        "Rig - Rig"
+    ].exists(), "Rig task is missing in content dialog"
+
+    # Search in the content dialog for Rig and make sure Model is not showing up anymore
+    app_dialog.root["Search All Files"].typeIn("Rig" "{ENTER}")
+    assert app_dialog.root.cells[
+        "Rig - Rig"
+    ].exists(), "Rig task should be visible in content dialog"
+    assert (
+        app_dialog.root.cells["Model - Model"].exists() is False
+    ), "Model task shouldn't be visible in content dialog"
+
+    # Remove text in the search field and make sure Modal task is back
+    app_dialog.root["Search All Files"].buttons.mouseClick()
+    assert app_dialog.root.cells[
+        "Model - Model"
+    ].exists(), "Model task should be visible in content dialog"
+
+    # Select Model task
+    app_dialog.root.outlineitems["Model"].mouseDoubleClick()
+    app_dialog.root.outlineitems["Model"][1].waitExist(timeout=30)
+    app_dialog.root.outlineitems["Model"][1].mouseClick()
+
+    # Validate content dialog
+    assert app_dialog.root.cells["Model - Model"].exists(), "Not on the right tasks"
+
+    # Create a new task and select it
+    app_dialog.root.buttons["+ New Task"].mouseClick()
+    app_dialog.root.dialogs["Shotgun: Create New Task"].waitExist(timeout=30)
+    app_dialog.root.dialogs["Shotgun: Create New Task"].textfields["Task Name"].pasteIn(
+        "New Task"
+    )
+    app_dialog.root.dialogs["Shotgun: Create New Task"].buttons["Create"].mouseClick()
+    app_dialog.root.outlineitems["New Task"].waitExist(timeout=30)
+
+    # Validate breadcrumb
+    assert app_dialog.root.captions[
+        "Assets * Character * Asset AssetAutomation * Step Model * Task Model"
+    ].exists(), "Breadcrumb not on the right task"
+
+    # Click on the back navigation button until back to the Assets context
+    for _i in range(0, 5):
+        # Click on the back navigation button
+        app_dialog.root.buttons["nav_prev_btn"].mouseClick()
+
+    # Validate Breadcrumb widget is only showing the project context
+    assert app_dialog.root.captions[
+        "Assets"
+    ].exists(), "Breadcrumb widget is not set correctly"
+
+    # Enable My Tasks Only and make sure Model task is not showing up anymore
+    app_dialog.root.checkboxes["My Tasks Only"].mouseClick()
+    assert app_dialog.root.outlineitems[
+        "New Task"
+    ].exists(), "New Task task should be visible"
+
+
+def test_shots_tab(app_dialog):
+    """
+    Shot tab UI validation
+    """
+    # Select the Shots tab
+    app_dialog.root.tabs["Shots"].mouseClick()
```

Comment:
At a quick glance it looks like some of this at the top is the same as the open tab. I wonder if some of this could be put in a fixture that could be shared?